examples/ensemble/plot_gradient_boosting_quantile.py

```diff
@@ -32,41 +36,87 @@ def f(x):
 y += noise
 y = y.astype(np.float32)
 
+# Split into train, test datasets.
+X_train, X_test, y_train, y_test = train_test_split(X, y)
+
 # Mesh the input space for evaluations of the real function, the prediction and
-# its MSE
+# its MSE.
 xx = np.atleast_2d(np.linspace(0, 10, 1000)).T
 xx = xx.astype(np.float32)
 
-alpha = 0.95
-
-clf = GradientBoostingRegressor(loss='quantile', alpha=alpha,
-                                n_estimators=250, max_depth=3,
-                                learning_rate=.1, min_samples_leaf=9,
-                                min_samples_split=9)
-
-clf.fit(X, y)
-
-# Make the prediction on the meshed x-axis
-y_upper = clf.predict(xx)
-
-clf.set_params(alpha=1.0 - alpha)
-clf.fit(X, y)
-
-# Make the prediction on the meshed x-axis
-y_lower = clf.predict(xx)
-
-clf.set_params(loss='ls')
-clf.fit(X, y)
-
-# Make the prediction on the meshed x-axis
-y_pred = clf.predict(xx)
+# %%
+# Let's fit a model based trained with MSE loss.
+gbr_mse = GradientBoostingRegressor(loss='ls', n_estimators=250, max_depth=3,
+                                    learning_rate=.1, min_samples_leaf=9,
+                                    min_samples_split=9)
+
+# %%
+# Let's fit this models and three others trained with
+# the quantile loss and alpha=0.05, 0.5, 0.95.
+# The models obtained for alpha=0.05 and alpha=0.95
+# produce a 95% confidence interval. The model trained
+# with alpha=0.5 produce the median regression:
+# there are the same number of targets above and below
+# the predicted values.
+gbrs = {
+    "q%1.2f" % alpha:
+        GradientBoostingRegressor(loss='quantile', alpha=alpha,
+                                  n_estimators=250, max_depth=3,
+                                  learning_rate=.1, min_samples_leaf=9,
+                                  min_samples_split=9)
+    for alpha in [0.05, 0.5, 0.95]
+}
+gbrs['MSE'] = gbr_mse
+
+for alpha, gbr in gbrs.items():
+    gbr.fit(X_train, y_train)
+
+# %%
+# Let's measure the models given :func:`mean_square_error` and
+# :func:`pinball_loss` metrics on the training datasets.
+results = []
+for name, gbr in sorted(gbrs.items()):
+    metrics = {'model': name}
+    y_pred = gbr.predict(X_train)
+    for alpha in [0.05, 0.5, 0.95]:
+        metrics["pbl=%1.2f" % alpha] = pinball_loss(
+            y_train, y_pred, alpha=alpha)
+    metrics['MSE'] = mean_squared_error(y_train, y_pred)
+    results.append(metrics)
+DataFrame(results).set_index('model')
+
+# %%
+# One column shows all models evaluated by the same metric.
+# The number of the diagonal is the metric evaluated on a
+# model trained to minimize this same metric. It is expected
+# to be the minimum value on this column.
```

Comment:
We could also add a note to state that since the noise of this synthetic data is Gaussian, we do not expect large tails and therefore the least squares model is a reasonable estimator of the median quantile and conversely, the median model also approximately minimize the MSE metric.